### DISCUSSION TITLE
GROOPS TEC maps

### DISCUSSION THREAD

**Discussion by laplace76** [groops-devs/groops#180] [Q&A] [answered]
Dear Developers,

The program GnssGriddedDataTimeSeries2Ionex converts GROOPS TEC maps in IONEX format, however it is not clear to me how to generate the GROOPS TEC maps: in the orbit estimation part of the cookbook, the ionosphere is decomposed in a stocastic part (the residual slant iono TEC) and in a deterministic part modelled as Spherical Harmonics expansion. However, the only parametrisation that writes a class is the IonosphereVTEC, and I understand in the related description such parametrisation is used only to separate ionopshere and biases in the single receiver case. So, in practice, is it possible to create IONEX files with the current GROOPS implementation?

**Comment by tmayerguerr** [comment, marked as answer]
Hi laplace76 ,

You are completly right, it is still on our TODO list to read and write TEC maps within [GnssProcessing](https://groops-devs.github.io/groops/html/GnssProcessing.html) (see [GnssParametrization::ionosphereMap](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#ionosphereMap)). Internal tests show that the estimated results do not look very satisfactory, so little bit more work is to do here. Unfortunately, I don't know at the moment when we will find the time.

regards
Torsten

**Reply by laplace76** [reply]
Many thanks for the prompt reply: as you are still in the investigation phase, it may be worth it to consider a slepian based model for the deterministic ionosphere, perhaps it may help to better fit the TEC in region such as Brazil, and to cope with the un-even spatial distribution of the GNSS measurements, in a global network. 
In the case of a regional network, modelling the slant delay with adjusted spherical harmonics (one expansion for each satellite) would provide better results.

Kind Regards
